ctx.go

```diff
@@ -1398,47 +1399,99 @@ func (c *DefaultCtx) Send(body []byte) error {
+	// This works differently than the github.com/gofiber/compression middleware.
+	// Optional. Default value false
+	Compress bool `json:"compress"`
+
+	// When set to true, enables byte range requests.
+	// Optional. Default value false
+	ByteRange bool `json:"byte_range"`
+
+	// When set to true, enables direct download.
+	//
+	// Optional. Default: false.
+	Download bool `json:"download"`
+
+	// Expiration duration for inactive file handlers.
+	// Use a negative time.Duration to disable it.
+	//
+	// Optional. Default value 10 * time.Second.
+	CacheDuration time.Duration `json:"cache_duration"`
+
+	// The value for the Cache-Control HTTP-header
+	// that is set on the file response. MaxAge is defined in seconds.
+	//
+	// Optional. Default value 0.
+	MaxAge int `json:"max_age"`
+}
 
 // SendFile transfers the file from the given path.
 // The file is not compressed by default, enable this by passing a 'true' argument
 // Sets the Content-Type response HTTP header field based on the filenames extension.
-func (c *DefaultCtx) SendFile(file string, compress ...bool) error {
+func (c *DefaultCtx) SendFile(file string, config ...SendFile) error {
 	// Save the filename, we will need it in the error message if the file isn't found
 	filename := file
 
+	route := c.Route()
+
+	var cfg SendFile
+	if len(config) > 0 {
+		cfg = config[0]
+	}
+
+	if cfg.CacheDuration == 0 {
+		cfg.CacheDuration = 10 * time.Second
+	}
+
 	// https://github.com/valyala/fasthttp/blob/c7576cc10cabfc9c993317a2d3f8355497bea156/fs.go#L129-L134
-	sendFileOnce.Do(func() {
-		const cacheDuration = 10 * time.Second
-		sendFileFS = &fasthttp.FS{
+	if route.sendFileFS == nil {
+		route.sendFileFS = &fasthttp.FS{
 			Root:                 "",
+			FS:                   cfg.FS,
 			AllowEmptyRoot:       true,
 			GenerateIndexPages:   false,
-			AcceptByteRange:      true,
-			Compress:             true,
+			AcceptByteRange:      cfg.ByteRange,
+			Compress:             cfg.Compress,
 			CompressedFileSuffix: c.app.config.CompressedFileSuffix,
-			CacheDuration:        cacheDuration,
+			CacheDuration:        cfg.CacheDuration,
+			SkipCache:            cfg.CacheDuration < 0,
 			IndexNames:           []string{"index.html"},
 			PathNotFound: func(ctx *fasthttp.RequestCtx) {
 				ctx.Response.SetStatusCode(StatusNotFound)
 			},
 		}
-		sendFileHandler = sendFileFS.NewRequestHandler()
-	})
+
+		if cfg.FS != nil {
+			route.sendFileFS.Root = "."
+		}
+
+		route.sendFileHandler = route.sendFileFS.NewRequestHandler()
+
+		maxAge := cfg.MaxAge
+		if maxAge > 0 {
+			route.sendFileCacheControlValue = "public, max-age=" + strconv.Itoa(maxAge)
+		}
+	}
 
 	// Keep original path for mutable params
 	c.pathOriginal = utils.CopyString(c.pathOriginal)
+
 	// Disable compression
-	if len(compress) == 0 || !compress[0] {
-		// https://github.com/valyala/fasthttp/blob/7cc6f4c513f9e0d3686142e0a1a5aa2f76b3194a/fs.go#L55
-		c.fasthttp.Request.Header.Del(HeaderAcceptEncoding)
+	if cfg.Compress {
```

Comment:
not sure about this

think the complication is activated by default in the fs handler and must be deactivated and not activated

Reply:
> not sure about this
> 
> think the complication is activated by default in the fs handler and must be deactivated and not activated

I think it's necessary to set encoding header to allow compression https://github.com/valyala/fasthttp/blob/7cc6f4c513f9e0d3686142e0a1a5aa2f76b3194a/fs.go#L64 otherwise it doesn't compress the served file

Reply:
@efectn if not already present, we can then have a unittest for the compression case and the non-compression case, which checks the response encoding and the response itself to see if it is compressed (by simply checking the content length).